djangoguestbook/static/js/guestbook/views/SignFormWidget.js

```diff
@@ -0,0 +1,58 @@
+define([
+	"dojo/_base/declare",
+	"dojo/_base/fx",
+	"dojo/_base/lang",
+	"dojo/dom-style",
+	"dojo/on",
+	"dijit/_WidgetBase",
+	"dijit/_TemplatedMixin",
+	"dijit/_WidgetsInTemplateMixin",
+	"dijit/form/ValidationTextBox",
+	"dijit/form/Button",
+	"dojo/text!./templates/SignFormWidget.html",
+	"/static/js/guestbook/views/models/GreetingStore.js"
+], function(declare, baseFx, lang, domStyle, on, _WidgetBase,
+			_TemplatedMixin, _WidgetsInTemplateMixin, ValidationTextBox, Button, template, GreetingStore){
+	return declare("guestbook.SignFormWidget", [_WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin ], {
+		// Our template - important!
+		templateString: template,
+		widgetsInTemplate: true,
+
+		// default value
+		guestbookName: "default_guestbook",
+
+		postCreate: function(){
+			this.inherited(arguments);
+			this.own(
+				on(this.signButtonNode, "click", lang.hitch(this, "_onclickSignBtn"))
+			);
+			this.GreetingStore = new GreetingStore();
+		},
+
+		_onclickSignBtn : function(){
+			this._signNewGreeting(this.GuestbookViewParent,
+				this.guestbookNameNode.value,
+				this.contentNode.value);
+		},
+
+		_signNewGreeting: function(guestbookView, guestbookName, greetingContent){
+			var _contentLength = greetingContent.length;
+			if (_contentLength > 0 && _contentLength <= 10){
```

Comment:
Em đã có xử lí trong template SignFormWidget rồi anh: 
<input type="text" value="" required="true"
			data-dojo-attach-point="contentNode"
			data-dojo-type="dijit/form/ValidationTextBox"
			data-dojo-props="pattern:'.{10}', invalidMessage:'Max of content is 10 chars.'" />
Code trong file js là để ngăn chặn lúc người dùng vẫn bất chấp nhập quá 10 kí tự và nhấn sign button